x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysTypesTests.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.plan.logical.command.sys;
+
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.sql.analysis.analyzer.Analyzer;
+import org.elasticsearch.xpack.sql.analysis.index.EsIndex;
+import org.elasticsearch.xpack.sql.analysis.index.IndexResolution;
+import org.elasticsearch.xpack.sql.analysis.index.IndexResolver;
+import org.elasticsearch.xpack.sql.expression.function.FunctionRegistry;
+import org.elasticsearch.xpack.sql.parser.SqlParser;
+import org.elasticsearch.xpack.sql.plan.logical.command.Command;
+import org.elasticsearch.xpack.sql.session.SqlSession;
+import org.elasticsearch.xpack.sql.type.DataType;
+import org.elasticsearch.xpack.sql.type.TypesTests;
+
+import java.sql.JDBCType;
+import java.util.List;
+import java.util.TimeZone;
+
+import static java.util.Arrays.asList;
+import static org.elasticsearch.action.ActionListener.wrap;
+import static org.mockito.Mockito.mock;
+
+public class SysTypesTests extends ESTestCase {
+
+    private final SqlParser parser = new SqlParser();
+
+    private Tuple<Command, SqlSession> sql(String sql) {
+        EsIndex test = new EsIndex("test", TypesTests.loadMapping("mapping-multi-field-with-nested.json", true));
+        Analyzer analyzer = new Analyzer(new FunctionRegistry(), IndexResolution.valid(test), TimeZone.getTimeZone("UTC"));
+        Command cmd = (Command) analyzer.analyze(parser.createStatement(sql), true);
+
+        IndexResolver resolver = mock(IndexResolver.class);
+        SqlSession session = new SqlSession(null, null, null, resolver, null, null, null);
+        return new Tuple<>(cmd, session);
+    }
+
+    public void testSysTypes() throws Exception {
+        Command cmd = sql("SYS TYPES").v1();
+
+        List<String> names = asList("BYTE", "LONG", "BINARY", "NULL", "INTEGER", "SHORT", "HALF_FLOAT", "SCALED_FLOAT", "FLOAT", "DOUBLE",
```

Comment:
Because the order is different than that of the enum declaration. We could add sorting to it but then it would be the same logic as the one in the code and I wanted to check that manually.